src/main/java/com/mihir/pipposample/Employee/EmployeeDao.java

```diff
@@ -0,0 +1,180 @@
+package com.mihir.pipposample.Employee;
+
+import com.google.inject.Inject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.ServerHeartbeatFailedEvent;
+import com.mongodb.event.ServerHeartbeatStartedEvent;
+import com.mongodb.event.ServerHeartbeatSucceededEvent;
+import com.mongodb.event.ServerMonitorListener;
+import org.bson.Document;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class EmployeeDao implements ServerMonitorListener {
+    private enum HEALTH_STATUS{
```

Comment:
Rename this to MongoDao, that implements a generic EmployeeDao
What if you've to write an implementation for mysql later
Then you'll just change the binding in Guice Module